atc/worker/artifact_source.go

```diff
@@ -90,7 +93,7 @@ func (source *artifactSource) StreamFile(
 		reader: tarReader,
 		closers: []io.Closer{
 			out,
-			zstdReader,
+			CloseWithError{zstdReader},
```

Comment:
Looks like the library [provides a convenience method](https://github.com/klauspost/compress/blob/e2d2089232ece4911272106ae65a7a06746ea38a/zstd/decoder.go#L396-L402) for this purpose, so we don't need our own wrapper

```suggestion
			zstdReader.IOReadCloser(),
```

Reply:
the `CloseWithError` wrapper is gone in the final changes. `zstdReader` is now a general interface `compressionReader`
https://github.com/concourse/concourse/pull/5398/files#diff-8643c2bdbcfa9111b8325ff14c713a65R104-R105